### PR TITLE
some more docs for CPG

### PR DESCRIPTION
**Pull request by bbrehm**
Link base.json for CPG format spec. Expand on some documentation strings.

Please mark other fields that need better docs.

### CODE REVIEW

**Comment by bbrehm** [comment]
Further questions:

"NAME", "FULLNAME": For example, "run" vs "io,shiftleft.dataflowenging.layers.dataflows.DataFlowRunner.run", right?

"SIGNATURE": What is the format? We need a few examples here, enough for people to generate and parse SIGNATURE strings.

File: codepropertygraph/src/main/resources/schemas/base.json [hunk 1]
```diff
@@ -15,8 +15,10 @@
...
+        {"id" : 4, "name": "ORDER", "comment": "General ordering property, such that the children of each AST-node are typically numbered from 1, ..., N (this is not enforced). The ordering has no technical meaning, but is used for pretty printing and OUGHT TO reflect order in the source code.", "valueType" : "int", "cardinality" : "one"},
+        {"id" : 40, "name": "ARGUMENT_INDEX", "comment": "AST-children of CALL nodes have an argument index, that is used to match call-site arguments with callee parameters. Explicit parameters are numbered from 1 to N, while index 0 is reserved for implicit self / this parameter. CALLs without implicit parameter therefore have arguments starting with index 1. AST-children of BLOCK nodes may have an argument index as well; in this case, the last argument index determines the return-value of a BLOCK expression.",
+
+          "valueType" : "int", "cardinality" : "one"},
 
         {"id" : 7, "name": "IS_EXTERNAL", "comment" : "Indicates that the construct is external, that is, it is referenced but not defined in the code", "valueType" : "boolean", "cardinality" : "one"},
```
[the remark this answers is not in the document]
Comment (ml86) [reply codepropertygraph/src/main/resources/schemas/base.json]: A method/function is consider external if we do not have its implementation at hand. Some in both cases they would be external.

**Comment by ml86** [comment]
"NAME" is the bare name of a METHOD.
"FULL_NAME" is a unique identifier of a METHOD.
"SIGNATURE" signatures are language specific and can be completely defined by the frontend. The backend does not touch the internals of the signature strings. It just compares them in order to match up call sites to methods.

"BINDS_TO" and "CONTAINS_NODE" are not intended to be used/provided by the frontends. It is an internal implementation detail.

**Comment by bbrehm** [comment]
>"NAME" is the bare name of a METHOD.
>"FULL_NAME" is a unique identifier of a METHOD.

So the `run` example is correct for NAME? 

Do I understand correctly that FULL_NAME is not really defined? In the example it ought to be "io.shiftleft.dataflowenging.layers.dataflows.DataFlowRunner.run", but `hex(sha3("io.shiftleft.dataflowenging.layers.dataflows.DataFlowRunner.run"))` would be a perfectly valid full name (because it is unique in scala, up to hash collisions)? And another perfectly valid FULL_NAME would be just an integer-index or pointer?

**Comment by ml86** [comment]
Yes the example is correct and yes FULL_NAME is not specified and a unique integer would be correct.
That said for FULL_NAME there is the expectation of a certain level of readability for the Ocular user.

**Comment by fabsx00** [comment]
(Please not, before merging, that the build is red on this one)

**Pull request merged**